crates/astria-core/src/primitive/v1/asset.rs

```diff
@@ -137,6 +252,13 @@ impl Id {
         Self(hash.into())
     }
 
+    fn from_denom_fmt(denom_fmt: &DenomFmt<'_>) -> Self {
```

Comment:
This is just a quick fix and should be replaced by a non-allocating implementation.

Also, the `Id::from_denom` above is should actually take a `Denom`, not a `str`.